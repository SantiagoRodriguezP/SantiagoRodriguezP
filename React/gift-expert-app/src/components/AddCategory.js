import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault(); // Prevenir el comportamiento por defecto
        if(inputValue.trim().length > 2){
            setCategories(catg => [inputValue, ...catg ])
            setInputValue('');
        }

        
    }
    return (
        <form onSubmit={handleSubmit}>
            <h1>{inputValue}</h1>
            <input
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}