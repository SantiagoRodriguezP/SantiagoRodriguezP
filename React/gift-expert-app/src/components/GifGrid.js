import React, { useState, useEffect, Fragment } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

import { useFecthGifs } from "../hooks/useFetchGifts"

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFecthGifs( category );

    

   /* const [images, setImages] = useState([]);

   
    //para no volver a ejecutar
     useEffect(() => {
        getGifs(category).then( (data) => {
            setImages(data);
        });
    }, [category]);*/

   
       getGifs();
    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> { category } </h3>

            { loading && <p className="animate__animated animate__flash">Loading</p> }

             <div className='card-grid'>
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            
            </div> 
        </>
    )
}
