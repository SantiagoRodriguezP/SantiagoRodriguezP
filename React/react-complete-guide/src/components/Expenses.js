import React, {useState} from 'react';
import ExpenseItem from './ExpenseItem';
import Card from './Card';
import '../index.css';
import ExpensesFilter from './ExpenseFilter';

const Expenses = ({ expe }) =>{
    const [filteredYear, setfilteredYear] = useState('2021');

    const filterChangeHandler = selectedYear => {
        setfilteredYear(selectedYear);
        console.log(selectedYear);
    }
    
    return (
        <div>

            <Card className='expenses'>
                <ExpensesFilter selected={filteredYear} onChangeFilter={filterChangeHandler} />
                {
                    expe.map((exp, id) => {
                        return (
                            <ExpenseItem
                                key={id}
                                title={exp.title}
                                amount={exp.amount}
                                date={exp.date}
                            />
                        )
                    })
                }

            </Card>
        </div>
    )
}

export default Expenses;