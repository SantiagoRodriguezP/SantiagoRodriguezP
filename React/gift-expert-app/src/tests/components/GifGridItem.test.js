import React from 'react'
import {shallow } from 'enzyme'
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en <GifGridItem/>', () => {

    const title = 'Titulo';
    const url = 'https://localhost';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    test('Debe mostrar el componente ', () => {
      

      // expect(wrapper).toMatchSnapshot();
    })

    test('debe tener un parrafo con el titulo ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    
    test('Debe tener la img igual a la URL y alt de los props ', () => {
        const img = wrapper.find('img');
        console.log(img.prop('src'));
        expect(img.prop('src')).toBe(url);
     //   expect(img.prop('src')).toBe(title);
    })
    
    test('debe tener animate_fadeIn ', () => {
        const div = wrapper.find('div');
      const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    
})
