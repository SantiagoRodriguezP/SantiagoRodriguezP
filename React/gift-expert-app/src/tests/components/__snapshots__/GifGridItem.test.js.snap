// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p>
     
     
  </p>
</div>
`;
